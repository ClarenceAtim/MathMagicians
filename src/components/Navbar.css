nav {
  background-color: #333;
  padding: 10px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
}

ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
  justify-content: center;
}

li {
  margin-right: 10px;
}

a {
  color: #fff;
  text-decoration: none;
  font-size: 16px;
  padding: 8px 12px;
  border-radius: 4px;
}

a.active {
  font-weight: bold;
  background-color: #666;
}

.app-container {
  padding-top: 60px;
}

@media screen and (max-width: 768px) {
  ul {
    flex-direction: column;
    align-items: center;
  }

  li {
    margin-bottom: 10px;
  }
}
