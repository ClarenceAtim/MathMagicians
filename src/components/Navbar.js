import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navbar.css';

function Navbar() {
  return (
    <nav>
      <div className="logo-container">
        <h1>Math Magicians</h1>
      </div>
      <ul>
        <li>
          <NavLink exact to="/" activeClassName="active">Home</NavLink>
        </li>
        <li>
          <NavLink to="/calculator" activeClassName="active">Calculator</NavLink>
        </li>
        <li>
          <NavLink to="/quote" activeClassName="active">Quote</NavLink>
        </li>
      </ul>
    </nav>
  );
}

export default Navbar;
