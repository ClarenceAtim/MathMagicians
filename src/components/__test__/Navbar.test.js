import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Navbar', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });

  it('activates the correct link when clicked', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );

    getByText('Home');
    getByText('Calculator');
    getByText('Quote');
  });
});
