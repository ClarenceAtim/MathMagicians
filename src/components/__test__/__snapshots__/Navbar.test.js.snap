// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<div>
  <nav>
    <div
      class="logo-container"
    >
      <h1>
        Math Magicians
      </h1>
    </div>
    <ul>
      <li>
        <a
          activeclassname="active"
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          class=""
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          class=""
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
